### PR TITLE
Ignite Workflow Lifecycle Management linked into GRPC Twin subscripti…

### PR DESCRIPTION
**Pull request by artnaseef**
…on.  Pinger attempts to run but returns errors.

### CODE REVIEW

File: ignite-worker/src/main/java/org/opennms/poc/ignite/worker/ignite/registries/MonitorRegistryImpl.java [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
 import com.savoirtech.eos.pattern.whiteboard.KeyedWhiteboard;
 import lombok.extern.slf4j.Slf4j;
-import org.apache.ignite.Ignite;
-import org.apache.ignite.services.Service;
 import org.opennms.poc.plugin.api.ServiceMonitor;
 import org.osgi.framework.BundleContext;
 
 @Slf4j
-public class MonitorRegistryImpl extends KeyedWhiteboard<String, ServiceMonitor> implements MonitorRegistry, Service {
+public class MonitorRegistryImpl extends KeyedWhiteboard<String, ServiceMonitor> implements MonitorRegistry {
```
Comment (mfrazier63) [inline ignite-worker/src/main/java/org/opennms/poc/ignite/worker/ignite/registries/MonitorRegistryImpl.java:9]: Good catch!

File: ignite-worker/src/main/java/org/opennms/poc/ignite/worker/workflows/WorkflowLifecycleManager.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package org.opennms.poc.ignite.worker.workflows;
+
+import org.opennms.poc.ignite.grpc.whiteboard.api.MessageListener;
+import org.opennms.poc.ignite.model.workflows.Workflows;
+
+/**
+ * Watcher for Workflow definitions responsible to manage worker lifecycles to ensure all Workflows are processed.
+ */
+public interface WorkflowLifecycleManager extends MessageListener<Workflows> {
```
Comment (mfrazier63) [inline ignite-worker/src/main/java/org/opennms/poc/ignite/worker/workflows/WorkflowLifecycleManager.java:9]: Is this just a marker interface?

File: ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml [hunk 3]
```diff
@@ -8,6 +8,7 @@
     <cm:default-properties>
       <cm:property name="useKubernetes" value="false"/>
       <cm:property name="kubernetes.service-name" value="poc-distributed-scheduling"/>
+      <cm:property name="workflowSchedulerThreadCount" value="20"/>
```
Comment (mfrazier63) [inline ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml:11]: Suggest: workflow.scheduler.threadcount

**Review by mfrazier63 — APPROVED** [review]
Just a couple nits

File: ignite-worker/src/main/java/org/opennms/poc/ignite/worker/workflows/WorkflowLifecycleManager.java (hunk 2 above)
Comment (artnaseef) [reply ignite-worker/src/main/java/org/opennms/poc/ignite/worker/workflows/WorkflowLifecycleManager.java:9]: It started as something more useful, but then `MessageListener` came along.  I considered to remove it, but the IMPL class is still called IMPL and is a sub-folder, so I left it for now.

Would be safe to simply remove it.  I think the idea of `WorkflowLifecycleManager` directly being a `MessageListener` is a little unclear, but adding another class to the mix feels like overkill too.

File: ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml (hunk 3 above)
Comment (artnaseef) [reply ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml:11]: I can change it.  Was thinking of using ENV-var-compatible naming to make the naming consistent when we use ENV vars with kubernetes, etc.

Thoughts?

Comment (mfrazier63) [reply ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml:11]: Seems like we always use "dot" notation in the properties files. I do find it a bit easier to parse for some reason

Comment (artnaseef) [reply ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml:11]: Agreed - but of course ENV variable names don't allow dots or dashes.  So if we start using K8S + ENV variables for all settings, we'll have 2 names for every setting - the internal property name and the ENV variable name.

Least-common-denominator FTW (sadly)...

**Pull request merged**